### ISSUE TITLE
Implement new Concept Exercise: error-handling

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#854]
This issue describes how to implement the `error-handling` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Error Handling in F#.

## Learning objectives

- Know when to use exceptions vs results vs options.
- Know what railway oriented programming is.
- Know how to use railway oriented programming, including custom operators.

## Out of scope

- Custom computation expression to deal with error-handling.
- Custom operators to deal with error-handling applicatively (like `<*>`).
- Converting error-handling to and from other types (like arrays of lists).

## Concepts

The Concepts this exercise unlocks are:

- `error-handling`: know when to use exceptions vs results vs options; know what railway oriented programming is; know how to use railway oriented programming, including custom operators.

## Prerequisites

This exercise's prerequisites Concepts are:

- `options`: know how to work with option values.
- `results`: know how to work with result values.
- `exceptions`: know how to work with exceptions.
- `operator-overloading`: how to define custom operators.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Railway oriented programming][railway-oriented-programming]: shows what railway oriented programming is and how to use it in F#.

### After

- [Railway oriented programming][railway-oriented-programming]: shows what railway oriented programming is and how to use it in F#.
- [Exceptions vs error-handling][exceptions-vs-result]: when to use exceptions vs error-handling.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/fsharp/reference/README.md#reference-docs
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[railway-oriented-programming]: https://fsharpforfunandprofit.com/posts/recipe-part1/#getting-started
[exceptions-vs-result]: https://danielwertheim.se/exceptions-vs-the-result-type-in-fsharp/
